packages/kbn-router-to-openapispec/src/oas_converter/kbn_config_schema/lib.ts

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import joi from 'joi';
+import { isConfigSchema, Type, metaFields } from '@kbn/config-schema';
+import { get } from 'lodash';
+import type { OpenAPIV3 } from 'openapi-types';
+import type { KnownParameters } from '../../type';
+import { isReferenceObject } from '../common';
+import { parse } from './parse';
+import { Context } from './post_process_mutations';
+
+export const getSharedComponentId = (schema: OpenAPIV3.SchemaObject) => {
+  if (metaFields.META_FIELD_X_OAS_REF_ID in schema) {
+    return schema[metaFields.META_FIELD_X_OAS_REF_ID] as string;
+  }
+};
+
+const removeSharedComponentId = (schema: OpenAPIV3.SchemaObject) => {
+  const { [metaFields.META_FIELD_X_OAS_REF_ID]: id, ...rest } = schema as any;
+  return rest;
+};
```

Comment:
NIT: return type is implicitly `any` due to the `schema as any` cast. Should we explicitly return `OpenAPIV3.SchemaObject`?

Reply:
Good catch!